x-pack/plugins/security_solution/common/endpoint/data_loaders/index_fleet_endpoint_policy.ts

```diff
@@ -106,18 +114,48 @@ export const indexFleetEndpointPolicy = usageTracker.track(
         version: packageVersion,
       },
     };
-    const packagePolicy = (await kbnClient
-      .request({
-        path: PACKAGE_POLICY_API_ROUTES.CREATE_PATTERN,
-        method: 'POST',
-        body: newPackagePolicyData,
-        headers: {
-          'elastic-api-version': API_VERSIONS.public.v1,
-        },
-      })
-      .catch(wrapErrorAndRejectPromise)) as AxiosResponse<CreatePackagePolicyResponse>;
 
-    response.integrationPolicies.push(packagePolicy.data.item as PolicyData);
+    const createPackagePolicy = async (): Promise<CreatePackagePolicyResponse> =>
+      kbnClient
+        .request<CreatePackagePolicyResponse>({
+          path: PACKAGE_POLICY_API_ROUTES.CREATE_PATTERN,
+          method: 'POST',
+          body: newPackagePolicyData,
+          headers: {
+            'elastic-api-version': API_VERSIONS.public.v1,
+          },
+        })
+        .catch(catchAxiosErrorFormatAndThrow)
+        .then((res) => res.data);
+
+    const started = new Date();
+    const hasTimedOut = (): boolean => {
+      const elapsedTime = Date.now() - started.getTime();
+      return elapsedTime > TimeoutsInMS.FIVE_MINUTES;
+    };
+
+    let packagePolicy: CreatePackagePolicyResponse | undefined;
+    log?.info(`Creating package policy for ${policyName}`);
+
+    while (!packagePolicy && !hasTimedOut()) {
+      packagePolicy = await retryOnError(
+        async () => createPackagePolicy(),
+        [...RETRYABLE_TRANSIENT_ERRORS, 'resource_not_found_exception'],
+        log
+      );
+
+      if (!packagePolicy) {
+        await new Promise((resolve) => setTimeout(resolve, TimeoutsInMS.TEN_SECONDS));
+      }
+    }
+
+    if (!packagePolicy) {
+      throw new Error(`Create package policy failed`);
+    }
+
+    log?.info(`Created package policy for ${policyName}`);
```

Comment:
Also here - I don't see the need for it to be `.info()`. maybe change it to `.verbose()` and also dump the `packagePolicy` document into it.

```typescript
log?.verbose(`Integration policy created:`, JSON.stringify(packagePolicy, null, 2))`
```

Reply:
Changed